data_pipeline_api/registry/access_upload.py

```diff
@@ -371,29 +403,31 @@ def upload_model_run(
     :param config_filename: file path to the configuration file
     :param model_config_filename: file path to the model configuration file
     :param submission_script_filename: file path to the submission script file
-    :param remote_uri: URI to the root of the storage for uploading
-    :param remote_uri_override: URI to the root of the storage that gets put into the data registry as the URI
-    :param storage_options: (key, value) pairs that are passed to the remote storage, e.g. credentials
-    :param data_registry_url: base url of the data registry
+    :param remote_options: (key, value) pairs that are passed to the remote storage, e.g. credentials
     :param token: personal access token
     :param text_file_table: If true, model_config and submission_script are uploaded to the text_file table in the data_registry
     """
-    remote_uri_override = remote_uri_override or remote_uri
     config_filename = Path(config_filename)
     with open(config_filename, "r") as cf:
         config = yaml.safe_load(cf)
-    data_directory = Path(config["data_directory"])
+
+    accessibility = _get_accessibility(config)
+    run_metadata = config["run_metadata"]
+
+    data_directory = Path(run_metadata[RunMetadata.data_directory])
     if not data_directory.is_absolute():
         data_directory = config_filename.parent / data_directory
-    run_id = config["run_id"]
-    config_yaml = config["config"]
-    accessibility = _get_accessibility(config)
-    namespace = config.get("namespace")
-    model_version_str = config_yaml["model_version"]
-    model_name = config_yaml["model_name"]
-    open_timestamp = config["open_timestamp"]
-    model_git_sha = config["metadata"]["git_sha"]
-    model_uri = config["metadata"]["uri"]
+    run_id = run_metadata[RunMetadata.run_id]
+    namespace = run_metadata.get(RunMetadata.default_output_namespace)
+    model_version_str = run_metadata[RunMetadata.model_version]
+    model_name = run_metadata[RunMetadata.model_name]
+    open_timestamp = run_metadata[RunMetadata.open_timestamp]
+    model_git_sha = run_metadata[RunMetadata.git_sha]
+    model_uri = run_metadata[RunMetadata.git_repo]
+    remote_uri = run_metadata[RunMetadata.remote_uri]
+    remote_uri_override = run_metadata[RunMetadata.remote_uri_override]
+    data_registry_url = run_metadata[RunMetadata.data_registry_url]
```

Comment:
and neither should this? use the get_data_registry_url function if not specified?